From a185bd2ee56c3057513c5c970b75e395eacbc54b Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Wed, 18 Jan 2023 14:37:52 +0100
Subject: [PATCH 250/841] feat(stm32mp2): update image IDs

To avoid taking too much place for authentication structures, reduce
the MAX_NUMBER_IDS by rewriting some IDs over unused IDs.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 33f30cb51671f441e211899f2faf1bbf025d9910 feat(stm32mp2): update image IDs]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: Ie835fba166ac35719316b48387eb623ee2bdc2c4
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/285745
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Lionel DEBIEVE <lionel.debieve@foss.st.com>
---
 plat/st/stm32mp2/include/plat_tbbr_img_def.h | 35 ++++++++++++++++----
 1 file changed, 29 insertions(+), 6 deletions(-)

diff --git a/plat/st/stm32mp2/include/plat_tbbr_img_def.h b/plat/st/stm32mp2/include/plat_tbbr_img_def.h
index 98604861e..8d319fa43 100644
--- a/plat/st/stm32mp2/include/plat_tbbr_img_def.h
+++ b/plat/st/stm32mp2/include/plat_tbbr_img_def.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021, STMicroelectronics - All Rights Reserved
+ * Copyright (c) 2021-2023, STMicroelectronics - All Rights Reserved
  *
  * SPDX-License-Identifier: BSD-3-Clause
  */
@@ -9,14 +9,37 @@
 
 #include <export/common/tbbr/tbbr_img_def_exp.h>
 
+/* Undef the existing values */
+#undef BKUP_FWU_METADATA_IMAGE_ID
+#undef FWU_METADATA_IMAGE_ID
+#undef FW_CONFIG_ID
+#undef ENC_IMAGE_ID
+#undef GPT_IMAGE_ID
+#undef NT_FW_CONFIG_ID
+#undef SOC_FW_CONFIG_ID
+#undef TB_FW_CONFIG_ID
+
+/* Define the STM32MP1 used ID */
+#define BKUP_FWU_METADATA_IMAGE_ID	U(1)
+#define FWU_METADATA_IMAGE_ID		U(2)
+#define FW_CONFIG_ID			U(8)
+#define ENC_IMAGE_ID			U(12)
+#define GPT_IMAGE_ID			U(16)
+#define NT_FW_CONFIG_ID			U(18)
+#define SOC_FW_CONFIG_ID		U(19)
+#define TB_FW_CONFIG_ID			U(20)
+#define STM32MP_CONFIG_CERT_ID		U(24)
+
 #if STM32MP_DDR_FIP_IO_STORAGE
-#define DDR_FW_ID		MAX_IMG_IDS_WITH_SPMDS
-#define MAX_IMG_WITH_DDR_IDS	MAX_IMG_IDS_WITH_SPMDS + 1
+#define DDR_FW_ID			U(27)
+/* Increase the MAX_NUMBER_IDS to match the authentication pool required */
+#define MAX_NUMBER_IDS			U(28)
+
 #else
-#define MAX_IMG_WITH_DDR_IDS	MAX_IMG_IDS_WITH_SPMDS
-#endif
+/* Increase the MAX_NUMBER_IDS to match the authentication pool required */
+#define MAX_NUMBER_IDS			U(27)
 
-#define MAX_NUMBER_IDS		MAX_IMG_WITH_DDR_IDS
+#endif
 
 #endif /* PLAT_TBBR_IMG_DEF_H */
 
-- 
2.47.2

